src/utils/filter-creators.ts

```diff
@@ -89,38 +84,52 @@ function createRangeFilter(fieldId: string, range: BoundedRange<number|Date>): F
 }
 
 /**
- * Creates a {@link Filter} that matches all results within a certain radius of the
- * given position.
+ * Creates a {@link FieldValueStaticFilter} that matches all results within a certain radius
+ * of the given position.
  *
  * @param position - The position and radius
- * @returns The newly created {@link Filter} for the radius of the position
+ * @returns The newly created {@link FieldValueStaticFilter} for the radius of the position
  *
  * @public
  */
-export function createNearMeFilter(position: NearFilterValue): Filter {
+export function createNearMeStaticFilter(position: NearFilterValue): FieldValueStaticFilter {
   return {
+    kind: 'fieldValue',
     fieldId: 'builtin.location',
     matcher: Matcher.Near,
     value: position
   };
 }
 
 /**
- * Creates a {@link CombinedFilter} by applying the specified {@link FilterCombinator}
- * to the two filters.
+ * Creates a {@link StaticFilter} by applying the specified {@link FilterCombinator}
+ * to the two static filters. Throws an error if an attempt is made to combine a
+ * conjunction static filter using {@link FilterCombinator.OR}.
  *
- * @param filterA - The first filter to be combined
- * @param filterB - The second filter to be combined
- * @param combinator - Specifies how the two filters should be joined
- * @returns The newly created {@link CombinedFilter}
+ * @param filterA - The first static filter to be combined
+ * @param filterB - The second static filter to be combined
+ * @param combinator - Specifies how the two static filters should be joined
+ * @returns The newly created {@link StaticFilter}
  *
  * @public
  */
-export function combineFilters(
-  filterA: FilterTypes,
-  filterB: FilterTypes,
-  combinator: FilterCombinator): CombinedFilter {
+export function combineStaticFilters(
+  filterA: StaticFilter,
+  filterB: StaticFilter,
+  combinator: FilterCombinator
+): StaticFilter {
+  if (combinator === FilterCombinator.OR) {
```

Comment:
Do we need runtime checking in addition to typescript? I haven't thought super deeply about it.

Reply:
Here, we need to have some checking so that typescript doesn't complain about this case. It gives an error if you try to combine 2 arbitrary StaticFilters together because they might be conjunctions being combined with a disjunction